common/corerequestcontext.go

```diff
@@ -17,6 +17,9 @@ func GetLogEntryFromContext(ctx context.Context) *logrus.Entry {
 
 // Deprecated: Use ServerParams.WithPkgLogger instead
 func GetLoggerFromContext(ctx context.Context) *logrus.Logger {
+	if getCoreContext(ctx) == nil {
```

Comment:
As a general approach, cache the result of a function rather than executing the function twice. It will:
1. Improve runtime performance.
2. Avoid issues if function may return a different value on a subsequent call.